File: 1-POO-Programacao-Orientada-a-Objetos/exercicios/src/HelloWorld.java
// Exemplo: Olá Mundo em Java com conceitos básicos de POO

public class HelloWorld {

    // Atributo da classe
    private String nome;

    // Construtor
    public HelloWorld(String nome) {
        this.nome = nome;
    }

    // Método getter (encapsulamento)
    public String getNome() {
        return nome;
    }

    // Método que exibe uma saudação
    public void saudar() {
        System.out.println("Olá, " + nome + "! Bem-vindo à POO com Java.");
    }

    public static void main(String[] args) {
        // Criando um objeto da classe HelloWorld
        HelloWorld obj = new HelloWorld("ADS");
        obj.saudar();
        System.out.println("Curso: " + obj.getNome());
    }
}

File: 1-POO-Programacao-Orientada-a-Objetos/exercicios/hello_world.py
# Exemplo: Conceitos básicos de POO em Python

# Classe base (abstração e encapsulamento)
class Animal:
    def __init__(self, nome, especie):
        self.__nome = nome       # atributo privado (encapsulamento)
        self.__especie = especie

    def get_nome(self):
        return self.__nome

    def get_especie(self):
        return self.__especie

    # Método que será sobrescrito (polimorfismo)
    def falar(self):
        return "..."

    def __str__(self):
        return f"{self.__especie}: {self.__nome}"


# Herança
class Cachorro(Animal):
    def __init__(self, nome):
        super().__init__(nome, "Cachorro")

    def falar(self):  # polimorfismo
        return "Au au!"


class Gato(Animal):
    def __init__(self, nome):
        super().__init__(nome, "Gato")

    def falar(self):  # polimorfismo
        return "Miau!"


# Criando objetos
animais = [Cachorro("Rex"), Gato("Mimi")]

for animal in animais:
    print(f"{animal} diz: {animal.falar()}")

